src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -170,7 +171,7 @@ public function edit($request)
         ])->renderWith($this->getTemplates());
 
         if ($request->isAjax()) {
-            return $return;
+            return $this->getResponseNegotiator($return)->respond($request);
```

Comment:
Doing this here means it applies for the save and publish actions, with a low chance of affecting custom actions people add.